### ISSUE TITLE
eks: Add L2 constructs for EKS Hybrid Nodes support

### ISSUE THREAD

**Issue by abhay-krishna** [aws/aws-cdk#32362]
### Describe the feature

On 12/1/2024, AWS announced the general availability of [EKS Hybrid Nodes](https://docs.aws.amazon.com/eks/latest/userguide/hybrid-nodes-overview.html), a new feature that allows customers to use their on-premises and edge infrastructure as nodes in Amazon EKS clusters. The L1 CloudFormation constructs for this change includes a new field `RemoteNetworkConfig` in the [Cluster](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-cluster.html) resource. I am requesting this field be supported in the EKS Cluster L2 construct so that customers can easily create Hybrid EKS clusters.

### Use Case

Creating Hybrid EKS clusters through L2 constructs without any direct dependency on L1 constructs.

### Proposed Solution

Incorporate RemoteNetworkConfig as a new field in the EKS Cluster L2 construct and add it to the `aws-cdk-lib/aws-eks` library.

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.171.1

### Environment details (OS name and version, etc.)

macOS Sequoia 15.1.1

**Comment by abhay-krishna** [comment]
Opened a pull request [here](https://github.com/aws/aws-cdk/pull/32370). Full steam ahead! 🚂

**Comment by boringgeek** [comment]
@abhay-krishna are you still looking to build this out?

**Comment by abhay-krishna** [comment]
> @abhay-krishna are you still looking to build this out? 

@boringgeek Yes! Currently waiting for reviews on [this PR](https://github.com/aws/aws-cdk/pull/32389).
